nixos/modules/services/networking/avahi-daemon.nix

```diff
@@ -202,22 +202,22 @@ in
 
       workstation = lib.mkOption {
         type = lib.types.bool;
-        default = false;
+        default = true;
         description = ''
           Whether to register a service of type "_workstation._tcp" on the local LAN.
         '';
       };
 
       domain = lib.mkOption {
         type = lib.types.bool;
-        default = false;
+        default = true;
         description = "Whether to announce the locally used domain name for browsing by other hosts.";
       };
     };
 
     nssmdns4 = lib.mkOption {
       type = lib.types.bool;
-      default = false;
+      default = true;
```

Comment:
This is the only option I have ever enabled on any of my clients. I think all the others are just announcing information into the network you are connecting to for no good reason and especially on public wifi we don't want to announce cpu information or why would we?

Reply:
> especially on public wifi we don't want to announce cpu information or why would we?

I agree that probably nobody cares about CPU information on a public Wi-Fi. However, I don't see where this does any harm.

Reply:
Tracking, mostly. Fingerprinting can involve basically any details of your system, so exposing as little of that as possible is valuable.

Reply:
What is the use case to publish those information? Why should it be enabled by default if we don't know why we enabled it?

Also do we really want to expand the output of avahi-browse -a with services and information that we coulnd't come up with a reason?

We probably want to change nssmdns4 to default to yes because that is the most common usecase for avahi and without it, it doesn't do much but I cannot follow why we should change all the other options.